CredentialProvider.Microsoft/Util/CertificateUtil.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using ILogger = NuGetCredentialProvider.Logging.ILogger;
+
+namespace NuGetCredentialProvider.Util
+{
+    internal static class CertificateUtil
+    {
+        public static X509Certificate2 GetCertificateBySubjectName(ILogger logger, string subjectName)
+        {
+            if (string.IsNullOrWhiteSpace(subjectName))
+            {
+                logger.Info(message: Resources.InvalidCertificateInput);
+                return null;
+            }
+
+            var locations = new List<StoreLocation> { StoreLocation.LocalMachine, StoreLocation.CurrentUser };
+            foreach (var location in locations)
+            {
+                var store = new X509Store(StoreName.My, location);
+                try
+                {
+                    store.Open(OpenFlags.ReadOnly);
+                    var cert = store.Certificates.Find(X509FindType.FindBySubjectDistinguishedName , subjectName, false);
+
+                    if (cert.Count > 0)
+                    {
+                        logger.Verbose(string.Format(Resources.ClientCertificateFound, subjectName));
+                        return cert[0];
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logger.Error(string.Format(Resources.ClientCertificateError, ex, ex.Message));
+                    continue;
+                }
+                finally
+                {
+                    store.Close();
+                }
+            }
+
+            logger.Info(string.Format(Resources.ClientCertificateSubjectNameNotFound, subjectName));
+            return null;
+        }
+
+        public static X509Certificate2 GetCertificateByFilePath(ILogger logger, string filePath)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                logger.Info(message: Resources.InvalidCertificateInput);
+                return null;
+            }
+
+            try
+            {
+                var certificate = new X509Certificate2(filePath);
+
+                if (certificate == null)
```

Comment:
Would expect an exception if the cert can't be allocated? was this a static method before?

Reply:
why throw vs return null or empty in the util layer? Regardless, if certificate information is provided and we fail to get a certificate, we should definitely have some user-friendly message about why - i have updated the build task message to a info instead of verbose to make it clearer.